File: app/src/main/java/com/zolax/testgooglemapsapi/MainActivity.java
package com.zolax.testgooglemapsapi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    Button mapButton;
    Intent mapIntent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mapButton = findViewById(R.id.mapButton);
        mapIntent = new Intent(MainActivity.this,MapsActivity.class);

        mapButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(mapIntent);
            }
        });

    }
}

File: app/src/main/java/com/example/testplacesapi/MainActivity.kt
package com.example.testplacesapi

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val mapButton = findViewById<Button>(R.id.map_button)
        val intent = Intent(this, MapsActivity::class.java)
        mapButton.setOnClickListener(View.OnClickListener {
            startActivity(intent)
        })
    }
}